scripts/analysisTools/w_mass_13TeV/run2Dsmoothing.py

```diff
@@ -774,16 +805,35 @@ def runSmoothing(inputfile, histname, outdir, step, args, effHist=None):
     )
     args = parser.parse_args()
 
-    if args.isolationDefinition == "iso04":
-        effSmoothFile = sfFolder + "efficiencies3D_rebinUt2.pkl.lz4"
-        #
-        inputRootFile = {
-            "iso": f"{sfFolder}iso3DSFVQTextended.root",
-            "isonotrig": f"{sfFolder}isonotrigger3DSFVQTextended.root",
-            "isoantitrig": f"{sfFolder}isofailtrigger3DSFVQTextended.root",
-            "triggerplus": f"{sfFolder}triggerplus3DSFVQTextended.root",
-            "triggerminus": f"{sfFolder}triggerminus3DSFVQTextended.root",
-        }
+    if not hasattr(args, "indir"):
+        args.indir =  data_dir + "/muonSF/"
+        logger.warning("Input folder for 3d efficiencies not specified. Using the default path.")
+
+
+    if args.era == "2016PreVFP":
+        logger.error("Run 2016 PreVFP is not currently available!")
+        quit()
+    elif args.era == "2016PostVFP":
+        sfFolder = data_dir + "/muonSF/"
+        inputRootFile = inputRootFile_2016PostVFP
+        if args.isolationDefinition == "iso04vtxAgn":
+            effSmoothFile = sfFolder + "intermediate_vtxAgnosticIso/efficiencies3D_rebinUt2_vtxAgnPfRelIso04.pkl.lz4"
+        else:
+            effSmoothFile = sfFolder + "effiencies3D_rebinUt2.pkl.lz4"
+    else:
+        if args.isolationDefinition == "iso04":
+            logger.error(f"Only 'iso04vtxAgn' is available for eras 2017 and 2018.")
+            quit()
```

Comment:
I would propose moving this exception as early as possible, if we know that some options are not compatible. Unless we want to use a different placeholder file but in general we would just quit.